### PR TITLE
Add extra syntax support for LIMIT & OFFSET

### PR DESCRIPTION
**Pull request by matriv**
## Summary of the changes / Why this improves CrateDB

To improve compatibility with PostgreSQL, add the exact syntax
which is supported by PostgreSQL for `LIMIT` and `OFFSET`:

```
LIMIT ALL
LIMIT NULL
OFFSET 10 ROW
OFFSET 10 ROWS
OFFSET NULL
```
Pre-work for: #11540 

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: libs/sql-parser/src/main/antlr/SqlBase.g4 [hunk 1]
```diff
@@ -134,8 +134,8 @@ query
 queryNoWith
     : queryTerm
       (ORDER BY sortItem (',' sortItem)*)?
-      (LIMIT limit=parameterOrInteger)?
-      (OFFSET offset=parameterOrInteger)?
+      (LIMIT (limit=parameterOrInteger | nullLiteral | ALL))?
```
Comment (mfussenegger) [inline libs/sql-parser/src/main/antlr/SqlBase.g4:137]: I think we could extend `parameterOrInteger` with `nullLiteral`. The code should/must already handle null values because a parameter could bind to `null` and it would reduce the amount of special expressions in the grammar

**Review by mfussenegger — APPROVED** [review]
Left a minor suggestion,  otherwise lgtm

**Review by seut — COMMENTED** [review]
good catch, thanks! added just some test related comments.

File: server/src/test/java/io/crate/analyze/UnionAnalyzerTest.java [hunk 2]
```diff
@@ -63,7 +63,7 @@ public void testUnion2Tables() {
             "union all " +
             "select id, name from users_multi_pk " +
             "order by id, 2 " +
-            "limit 10 offset 20");
+            "limit 10 offset 20 row");
```
[the remark this answers is not in the document]
Comment (matriv) [reply server/src/test/java/io/crate/analyze/UnionAnalyzerTest.java:66]: There is no concrete argument, I just thought to add it randomly here and there.
I can remove it and let it only in the `SelectStatementAnalyzerTest`.

[the remark this answers is not in the document]
Comment (matriv) [reply server/src/test/java/io/crate/analyze/UnionAnalyzerTest.java:66]: It was more of a thought: `let's use it here and there, so it's visible in the future that it's supported`.

File: libs/sql-parser/src/main/antlr/SqlBase.g4 (hunk 1 above)
Comment (mfussenegger) [reply libs/sql-parser/src/main/antlr/SqlBase.g4:137]: It strictly speaking isn't supported from a semantic level, but a user could use a `?` and then pass a `NULL` value, so allowing `NULL` directly in the parser shouldn't break things, as some component after the parser will have to deal with it anyways.

File: server/src/test/java/io/crate/analyze/UnionAnalyzerTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (seut) [reply server/src/test/java/io/crate/analyze/UnionAnalyzerTest.java:66]: I see, thanks for the clarification. I'd prefer not to randomly add it/optional stuff which does not change anything and is just ignored. But no strong opinion.

File: libs/sql-parser/src/main/antlr/SqlBase.g4 (hunk 1 above)
Comment (matriv) [reply libs/sql-parser/src/main/antlr/SqlBase.g4:137]: ok, let me try with allowing null everywhere.

File: server/src/test/java/io/crate/analyze/UnionAnalyzerTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (matriv) [reply server/src/test/java/io/crate/analyze/UnionAnalyzerTest.java:66]: Ok, no worries, I'll remove

File: libs/sql-parser/src/main/antlr/SqlBase.g4 (hunk 1 above)
Comment (matriv) [reply libs/sql-parser/src/main/antlr/SqlBase.g4]: As discussed, I'll merge it with the change to allow `nullLiteral` for `parameterOrInteger` and open another PR to add missing null checks for the `CLUSTERED INTO` and the shard rerouting commands.

**Comment by matriv** [comment]
@mfussenegger Would be great to give it another quick look, (fixed the analyzer test, regarding the expected `null` or null Literal)

**Review by mfussenegger — APPROVED** [review]
